src/server/auth.ts

```diff
@@ -53,3 +54,43 @@ export default async function initializePatronTokenAuth(req: NextRequest) {
 
   return patronTokenResponse
 }
+
+/**
+ * Creates and returns redirect url from initializePatronTokenAuth() response. Call this function only with an un-authenticated patron,
+ * i.e., patronTokenResponse.isTokenValid must be false.
+ */
+export function getLoginRedirect(req) {
+  const protocol = "http"
+  const hostname = req.headers["host"]
+  const originalUrl = BASE_URL + req.url
+  const fullUrl = encodeURIComponent(`${protocol}://${hostname}${originalUrl}`)
+  const redirect = `${
+    appConfig.externalUrls.loginUrl[appConfig.environment]
+  }?redirect_uri=${fullUrl}`
+  return redirect
+}
+
+/**
+ * Creates redirect to log out user, then return user to their current page.
+ */
+export const useLogoutRedirect = () => {
+  const baseUrl = "/research/research-catalog"
+  // Will send user back to prod if below doesn't work
+  const [redirect, setRedirect] = useState(`https://www.nypl.org${baseUrl}`)
```

Comment:
Use the `BASE_URL` constant, similar to how you used it in the previous function. And also on line 87.